application/src/test/java/org/opentripplanner/model/TripTimeOnDateTest.java

```diff
@@ -50,4 +58,42 @@ void isRecordedStop() {
 
     assertTrue(subject.isRecordedStop());
   }
+
+  @Test
+  void testFromTripTimesWithScheduleFallback() {
+    var testModel = TimetableRepositoryForTest.of();
+    var trip = TimetableRepositoryForTest.trip("123").build();
+    var siteRepository = testModel.siteRepositoryBuilder().build();
+    var timetableRepository = new TimetableRepository(siteRepository, new Deduplicator());
+    var tripTimes = ScheduledTripTimes.of()
+      .withTrip(trip)
+      .withDepartureTimes(new int[] { 0, 1 })
+      .build();
+    var tripPattern = testModel
+      .pattern(TransitMode.BUS)
+      .withScheduledTimeTableBuilder(builder -> builder.addTripTimes(tripTimes))
+      .build();
+    timetableRepository.addTripPattern(tripPattern.getId(), tripPattern);
+    timetableRepository.index();
+    var timetableSnapshot = new TimetableSnapshot();
+    timetableSnapshot.commit();
+    var transitService = new DefaultTransitService(timetableRepository, timetableSnapshot);
+    var serviceDate = LocalDate.of(2025, 1, 1);
+    // Construct a timetable which definitely does not contain this trip, because it is empty.
+    Timetable timetable = Timetable.of()
+      .withTripPattern(tripPattern)
+      .withServiceDate(serviceDate)
+      .build();
+    Instant midnight = ServiceDateUtils.asStartOfService(serviceDate, ZoneIds.HELSINKI).toInstant();
+    var tripTimeOnDates = TripTimeOnDate.fromTripTimesWithScheduleFallback(
+      timetable,
+      trip,
+      serviceDate,
+      midnight
+    );
+    for (var tripTimeOnDate : tripTimeOnDates) {
+      assertNull(tripTimeOnDate.getServiceDay());
+      assertEquals(tripTimeOnDate.getServiceDayMidnight(), TripTimeOnDate.UNDEFINED);
```

Comment:
Should we also validate that these triptimes have some "valid" data that can be used for something?